### PR TITLE
refactor: sampling

### PR DESCRIPTION
**Pull request by synchon**
This PR address #817 .
- Uses `Configuration.tolerance` to set `bounds_tol` and `feasibility_tol`. Should it use `model.tolerance` instead?
- Should it replace the `processes` in `OptGPSampler` with `Configuration.processes`?

### CODE REVIEW

**Comment by synchon** [comment]
The tests will pass once #805 is merged.

**Comment by synchon** [comment]
Reading the tolerance from the model will definitely be a better decision and I had that in my mind. That will also be a significant change to the way sampling was handled till now.

Another point is that there are two tolerances: feasibility tolerance and bounds tolerance, and both used the same values till now. Should that be the way as it is and be set to the model tolerance?

The ability to change the tolerance will also allow us to add a few more tests. Is that something we want to do?

File: cobra/test/test_sampling/test_sampling.py [hunk 1]
```diff
@@ -9,145 +9,75 @@
...
-        sample(model, 1, method="schwupdiwupp")
 
-
-def test_validate_wrong_sample(achr, model):
-    """Test sample correctness."""
-    s = achr.sample(10)
-    s["hello"] = 1
     with pytest.raises(ValueError):
-        achr.validate(s)
+        sample(model, 1, method="schwupdiwupp")
```
Comment (Midnighter) [inline cobra/test/test_sampling/test_sampling.py]: :rofl:

**Comment by synchon** [comment]
Did a recursive isort on `sampling/` and `test_sampling/`. Not sure why the test is still failing. :(

**Comment by Midnighter** [comment]
You may have to rebase or merge in the latest config changes from devel.

**Comment by synchon** [comment]
Should be good to go.

**Comment by cdiener** [comment]
Nice, looks great.

**Pull request merged**